### PR TITLE
Menu component using a portal to render the dropdown

### PR DESCRIPTION
**Pull request by wyunreal**

### CODE REVIEW

File: src/__screenshot_tests__/menu-screenshot-test.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import {openStoryPage, screen} from '../test-utils';
```
Comment (Winde) [inline src/__screenshot_tests__/menu-screenshot-test.tsx:1]: Worth it to test a menu inside one of the components that were interacting with the visibility: hidden? So that any future changes to menu or data-card,etc.. will not have similar wrong interaction?

**Review by Winde — APPROVED** [review]

File: src/menu.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
 import * as React from 'react';
 import classnames from 'classnames';
-import {assignInlineVars} from '@vanilla-extract/dynamic';
 import {ESC, TAB} from './utils/key-codes';
 import {cancelEvent, getPrefixedDataAttributes} from './utils/dom';
 import Overlay from './overlay';
 import * as styles from './menu.css';
-import {sprinkles} from './sprinkles.css';
+import {Portal, useWindowSize} from '.';
```
Comment (sergivillar) [inline src/menu.tsx:7]: inside mistica I think that we always import things from the component's file `./Portal` and not from the index

File: src/menu.tsx [hunk 3]
```diff
@@ -54,42 +62,50 @@ const Menu: React.FC<MenuProps> = ({renderTarget, renderMenu, width, position =
...
+        const availableSpaceOnBottom = windowSize.height - bottomTarget - MARGIN_THRESHOLD;
+        const availableSpaceOnTop = topTarget - MARGIN_THRESHOLD;
+        const menuFitsOnBottom = availableSpaceOnBottom > heightMenu;
+        const menuFitsOnTop = availableSpaceOnTop > heightMenu;
+
+        if (menuFitsOnBottom) {
             setItemsComputedProps({
-                top: '100%',
+                left: leftDirection,
+                top: `${bottomTarget}px`,
```
Comment (sergivillar) [inline src/menu.tsx:80]: unneeded pc units, this could be just numbers

File: src/menu.tsx [hunk 4]
```diff
@@ -54,42 +62,50 @@ const Menu: React.FC<MenuProps> = ({renderTarget, renderMenu, width, position =
...
+
+        const leftDirection =
+            position === 'left' ? `${leftTarget}px` : `${leftTarget + widthTarget - width}px`;
+
+        const availableSpaceOnBottom = windowSize.height - bottomTarget - MARGIN_THRESHOLD;
+        const availableSpaceOnTop = topTarget - MARGIN_THRESHOLD;
+        const menuFitsOnBottom = availableSpaceOnBottom > heightMenu;
+        const menuFitsOnTop = availableSpaceOnTop > heightMenu;
+
+        if (menuFitsOnBottom) {
```
Comment (sergivillar) [inline src/menu.tsx:77]: previously we had 3 branches to draw the menu, now we have 4, is that new branch (if-else) needed?

File: src/menu.tsx (hunk 2 above)
Comment (atabel) [reply src/menu.tsx:7]: yes please, this avoids circular dependencies

File: src/menu.tsx [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
 import {ESC, TAB} from './utils/key-codes';
 import {cancelEvent, getPrefixedDataAttributes} from './utils/dom';
 import Overlay from './overlay';
 import * as styles from './menu.css';
-import {sprinkles} from './sprinkles.css';
+import {Portal, useWindowSize} from '.';
 
 import type {DataAttributes} from './utils/types';
 
-const MAX_HEIGHT_DEFAULT = 416;
```
Comment (atabel) [inline src/menu.tsx:12]: I think the Menu should have a max height to avoid edge cases where there are to much items in the menu to fit in the screen (in small screens) or showing a huge list (in large screens). In the select, for example, we have a max height

**Review by atabel — APPROVED** [review]

File: src/__screenshot_tests__/menu-screenshot-test.tsx (hunk 1 above)
Comment (wyunreal) [reply src/__screenshot_tests__/menu-screenshot-test.tsx:1]: Update: added a story and a test case for this.

Comment (atabel) [reply src/__screenshot_tests__/menu-screenshot-test.tsx:1]: It's a good idea to test it here, yes

**Review by atabel — APPROVED** [review]

File: src/menu.tsx [hunk 6]
```diff
@@ -148,35 +166,39 @@ const Menu: React.FC<MenuProps> = ({renderTarget, renderMenu, width, position =
...
+                                ...assignInlineVars({
+                                    ...(itemsComputedProps
+                                        ? {
+                                              [styles.vars.top]: itemsComputedProps.top,
+                                              [styles.vars.bottom]: itemsComputedProps.bottom,
+                                              [styles.vars.left]: `${itemsComputedProps.left}px`,
+                                              [styles.vars.transformOrigin]:
+                                                  itemsComputedProps.transformOrigin,
+                                              [styles.vars.maxHeight]: `${itemsComputedProps.maxHeight}px`,
+                                              [styles.vars.width]: `${width}px`,
```
Comment (sergivillar) [inline src/menu.tsx:191]: some units with `px` and others as numbers

File: src/menu.css.ts [hunk 7]
```diff
@@ -1,40 +1,41 @@
...
 const width = createVar();
-const transformOrigin = createVar();
 const maxHeight = createVar();
+const transformOrigin = createVar();
 
 export const vars = {
     top,
     bottom,
-    right,
+    left,
```
Comment (sergivillar) [inline src/menu.css.ts:15]: is this change needed?

File: src/menu.css.ts [hunk 8]
```diff
@@ -1,40 +1,41 @@
...
         bottom,
-        right,
+        left,
         width,
-        transformOrigin,
         maxHeight,
+        transformOrigin,
+        display: 'flex',
+        flexDirection: 'column',
+        position: 'absolute',
```
Comment (sergivillar) [inline src/menu.css.ts:36]: move to sprinkles (position was already there before), is display flex needed?

File: src/menu.tsx (hunk 6 above)
Comment (wyunreal) [reply src/menu.tsx:191]: Yes, and its intentional, I prefer to have a number when I want to restrict the type only to numbers, because I dont need "auto" or "100%".

File: src/menu.css.ts (hunk 7 above)
Comment (wyunreal) [reply src/menu.css.ts:15]: Yes, because its easier to understand by anyone a left oriented coordinate system.

File: src/menu.css.ts (hunk 8 above)
Comment (wyunreal) [reply src/menu.css.ts:36]: Display flex is needed to make the menu items adopt all the horizontal space available for the menu.

Comment (wyunreal) [reply src/menu.css.ts:36]: Moving these props to sprinkles ;)

**Review by sergivillar — APPROVED** [review]
